roachtest: check in CI configs

tbg opened the pull request:
The configurations for our nightly CI runs are maintained ad-hoc in
TeamCity. In particular, there's a separate shell script for AWS and one
for GCE. This is error-prone and hard to maintain. One argument we've
had in the past for keeping it on TC is that it's easier to make tweaks,
but I'd posit that we can just paste the script we're tweaking into CI
while tweaking (ideally in a scratch copy of the job), or we can find
ways to run the nightly off a custom branch (which is difficult right
now due to the Publish Bleeding Edge dependency).

The shell scripts were copied without modifications. Obviously they
should be consolidated as a follow-up step.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/44852)
<!-- Reviewable:end -->

tbg committed:
roachtest: check in CI configs

The configurations for our nightly CI runs are maintained ad-hoc in
TeamCity. In particular, there's a separate shell script for AWS and one
for GCE. This is error-prone and hard to maintain. One argument we've
had in the past for keeping it on TC is that it's easier to make tweaks,
but I'd posit that we can just paste the script we're tweaking into CI
while tweaking (ideally in a scratch copy of the job), or we can find
ways to run the nightly off a custom branch (which is difficult right
now due to the Publish Bleeding Edge dependency).

The shell scripts were copied without modifications. Obviously they
should be consolidated as a follow-up step.

Release note: None

bobvawter reviewed (approved):
LGTM I'm a big fan of moving as much as possible out of TC (or any other just-tweak-it UI) and into something that's properly version-controlled.

> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/44852#-:-M-or1En1ZUWR0oJro-1:b-aebs2s)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

tbg committed:
build: update teamcity-nightly-roachtest-*

These are changes that happened while I had the PR to check in the
configs into VCS open.

Release note: None

tbg commented:
Finally got back to this. Added a commit picking up new changes to the source scripts (on TC). Once this is merged, I will replace the TC script with basically this

```
#/bin/bash
set -euxo pipefail

script_branch="${BRANCH_OVERRIDE-master}"
echo "Using build script from ${script_branch}"
git checkout "origin/${script_branch}"
git pull --ff-only "origin/${script_branch}"

CLOUD=aws ./build/teamcity-nightly-roachtest.sh
#       ^--- or gce, respectively
```

I'll use https://github.com/cockroachdb/dev-inf/issues/42#issuecomment-598756815 to track this. Once this is in I will fix up the artifacts as described here.

tbg commented:
bors r=bobvawter
